conda/recipes/cudf/meta.yaml

```
@@ -60,10 +60,10 @@ requirements:
     - protobuf >=4.21.6,<4.22
     - python
     - typing_extensions
-    - pandas >=1.0,<1.6.0dev0
+    - pandas >=1.3,<1.6.0dev0
     - cupy >=9.5.0,<12.0.0a0
     - numba >=0.56.2
-    - numpy
+    - numpy >=1.21
```

Current versions of numba [require `numpy<1.24`](https://github.com/numba/numba/issues/8615#issuecomment-1386886044) (until numba 0.57 is released with [this fix](https://github.com/numba/numba/pull/8691)). We're getting this pinning [from the upstream feedstock](https://github.com/conda-forge/numba-feedstock/blob/9662fd9e630719122c598edde702d96fa328bf94/recipe/meta.yaml#L67) but this pinning doesn't work if the user installs numba another way.

@jayrodge ran into this locally when installing RAPIDS 23.02 on WSL, perhaps with a numba from some source other than `conda-forge`.

Should we actively prevent that problem by pinning like this until numba 0.57 is released?
```suggestion
    - numpy >=1.21,<1.24
```

---

Even if we pin `numpy` in our feedstock, if the user install `numba` or as a matter of fact `numpy` in another we wouldn't be able to prevent these kinds of breakages no? I'm only hesitant on having an upper bound on `numpy` because we don't use it's internal APIs and this will have a maintenance overhead of tracking numpy releases and unpinning them manually. Whereas leaving it unpinned on the upper bound could help us detect breakages in our CI and have it fixed ASAP.

---

I understand the hesitation. I just wanted to pose the idea. Let’s leave it alone.